ttnn/cpp/ttnn/operations/data_movement/split/split.cpp

```diff
@@ -4,20 +4,90 @@
 
 
 #include "ttnn/common/constants.hpp"
+#include "ttnn/operations/core/core.hpp"
 #include "ttnn/run_operation.hpp"
 #include "device/split_op.hpp"
 #include "ttnn/cpp/ttnn/operations/data_movement/reshape/reshape.hpp"
 #include "ttnn/operations/data_movement/transpose/transpose.hpp"
+#include "ttnn/tensor/types.hpp"
 #include "ttnn/operations/data_movement/split/split.hpp"
+#include "ttnn/operations/data_movement/slice/slice.hpp"
 
 
 namespace ttnn::operations::data_movement {
 
 
 namespace detail {
 
-    std::vector<Tensor> impl_split_last_dim_two_chunks_tiled(const Tensor &input_tensor, const MemoryConfig &mem_config) {
+    std::vector<Tensor> split_dim_n_chunks_rm(const Tensor &input_tensor, int dim, int num_splits, const MemoryConfig &mem_config) {
+        TT_FATAL(input_tensor.get_layout() == Layout::ROW_MAJOR, "This op only supports row major tensors.");
+        TT_FATAL(input_tensor.get_shape()[dim] % num_splits == 0, "Split dimension must be divisible by num_splits.");
+        auto input_shape = input_tensor.get_shape();
+        auto input_rank = input_shape.size();
+
+        const bool on_host = input_tensor.storage_type() == StorageType::OWNED || input_tensor.storage_type() == StorageType::BORROWED;
+        std::optional<Device *> dev = on_host ? std::nullopt : std::make_optional(input_tensor.device());
```

Comment:
please use clear names, no need to save on 3 characters here
```suggestion
        std::optional<Device *> device = on_host ? std::nullopt : std::make_optional(input_tensor.device());
```